### ISSUE TITLE
Warn when no space is left

### ISSUE THREAD

**Issue by HeikoTietze** [sddm/sddm#742]
Updated Arch Linux today and wasn't able to start LXQt (some flickers and back to SDDM) or Plasma (hangs completely) afterwards. Took me a while to figure out that there was no space left on root. 

The request is to show an appropriate warning.

**Comment by eliasp** [comment]
Why would sddm refuse to start when there's no space left on the rootfs?
Shouldn't all runtime data end up in a tempfs?
When it is unable to update `/var/lib/sddm/state.conf` it should possibly fail gracefully…

**Comment by HeikoTietze** [comment]
It's not sddm that doesn't start but the DE. And I think sddm should warn if exec or whatever is being called will likely not work.

**Comment by sten0** [comment]
I wonder if code can be borrowed from the Plasma Desktop "Hard Disk Space Usage" widget, and then make this widget available for SDDM themes?  I don't know if it will issue a warning, but it would be immediately obvious when a system is about to run out of free space, before attempting to login.

**Comment by jasontibbitts** [comment]
Is there a hook further on in the stack for hanging checks like this?  I suppose that would be something in the Xsession file but with KDM at least that's actually part of the display manager.

**Comment by davidedmundson** [comment]
Yes. See Sddm manual for our hooks, the session you're logging into will also have relevant hooks. (KDE does anyway)
